tests/baselines/reference/declFileRestParametersOfFunctionAndFunctionType.js

```diff
@@ -27,10 +27,10 @@ var f6 = function () { return [10]; };
 declare function f1(...args: any[]): void;
 declare function f2(x: (...args) => void): void;
 declare function f3(x: {
-    (...args): void;
+    (...args: any[]): void;

```

Comment:
Why are we adding `: any[]` here when we are not adding it in `f2`? Both `f2` and `f3` define `args` without a type.

Reply:
I reread the PR summary and I see that this is an expected change, but it also seems that we are still incorrect when emitting `f2` above (per your reasoning).

Reply:
Turns out I never visited `FunctionType` or `ConstructorType` nodes, since under no other scenario do we need to edit them prior to emit. I've now added them just for this. 😉